### ISSUE TITLE
Daily Usage Ideas

### ISSUE THREAD

**Issue by iamjohnford** [vidoardes/dex#41]
Thanks so much for this amazing tool and the energy you spend maintaining it. I've been using the app multiple times a week for a number of weeks now and wanted to share some observations.

**Finding a Pokemon**
After a lucky trade, or catching a new shiny, I like to open the app and update my dex. However, it takes a good bit of time to scroll through and find the pokemon I want to update. I usually keep scrolling and use the browser search to check if the pokemon has shown up as I scroll. Even if I do some filtering it's still cumbersome to scroll and wait for the infinite scroll to load. There also seems to be an occasional issue where the data doesn't reload when changing/clearing filters but I haven't figured out the exact steps to replicate it consistently.

A text box to filter by name or number could help solve this. I pretty much always know the pokemon name so if I could search for that it would be amazing and speed up this process.

**Loading Time**
I'm sure specific decisions were made with the infinite scroll and loading only a small set of data on the page at any time. However, the entire dex could probably be loaded at once without seeing a performance hit. There seems to be a lot of extra data in the JSON response that's not being used on the page which could be removed to keep the response data smaller. The images could still be eager loaded when scrolling and filtering/searching could be done with the front-end JS to keep it really snappy.

You may have already thought about all of this but I just wanted to share the few things that I think would make some of these app interactions even better.

**Comment by vidoardes** [comment]
Both of these are good ideas. The reason the lazy loading was implemented wasn't because of the images, it was the sheer amount of DOM elements on the page causing it to lag massively on first load, given that there are now over 500 pokemon, which meant 7500+ DOM elements.

The searchable drop for pokemon is something I have implemented in the admin area for updating shinies and raid bosses etc. but I am in the middle of a rewrite to totally remove the front end to a react app and make this repo a proper restful API that has alternate uses. I can check out an old branch and see if I can add this to the old map in the meantime, but it might be while before I can get back to this.
